### PR TITLE
Refactor fetch serial publication tests

### PR DESCRIPTION
**Pull request by monaullah**
Some follow ups after previous PR:
- Replace `JournalDto` and `SeriesDto` with `SerialPublictaionDto` all places
- Move common tests for all types (journal, series and publisher) to `FetchByIdentifierAndYearHandlerTest`
- Move common tests for types journal and series to `FetchSerialPublicationByIdentifierAndYearHandlerTest`

### CODE REVIEW

File: src/test/java/no/sikt/nva/pubchannels/handler/fetch/serialpublication/FetchSerialPublicationByIdentifierAndYearHandlerTest.java [hunk 1]
```diff
@@ -1,313 +1,108 @@
...
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.MethodSource;
-import org.junit.jupiter.params.provider.ValueSource;
 import org.zalando.problem.Problem;
 
-public class FetchSerialPublicationByIdentifierAndYearHandlerTest extends FetchByIdentifierAndYearHandlerTest {
+public class FetchSerialPublicationByIdentifierAndYearHandlerTest extends
+                                                                  no.sikt.nva.pubchannels.handler.fetch.FetchSerialPublicationByIdentifierAndYearHandlerTest {
```
Comment (monaullah) [inline src/test/java/no/sikt/nva/pubchannels/handler/fetch/serialpublication/FetchSerialPublicationByIdentifierAndYearHandlerTest.java:34]: Not a fan of two classes with same name `FetchSerialPublicationByIdentifierAndYearHandlerTest`. What should i call the abstract class `FetchSerialPublicationByIdentifierAndYearHandlerTest`?

Comment (LarsV123) [reply src/test/java/no/sikt/nva/pubchannels/handler/fetch/serialpublication/FetchSerialPublicationByIdentifierAndYearHandlerTest.java:34]: Maybe just `FetchPublicationByIdentifierAndYearHandlerTest`?

File: src/main/java/no/sikt/nva/pubchannels/handler/create/journal/CreateJournalHandler.java [hunk 2]
```diff
@@ -49,7 +49,7 @@ protected JournalDto processInput(CreateJournalRequest input, RequestInfo reques
         var newJournal = (ThirdPartySerialPublication) publicationChannelClient.getChannel(JOURNAL,
                                                                                            response.pid(),
                                                                                            year);
-        var journalDto = JournalDto.create(constructBaseUri(JOURNAL_PATH_ELEMENT), newJournal, year);
+        var journalDto = SerialPublicationDto.create(constructBaseUri(JOURNAL_PATH_ELEMENT), newJournal, year);
```
Comment (LarsV123) [inline src/main/java/no/sikt/nva/pubchannels/handler/create/journal/CreateJournalHandler.java:52]: nitpick: Misleading name now?

File: src/test/java/no/sikt/nva/pubchannels/handler/fetch/BaseFetchSerialPublicationByIdentifierAndYearHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        var actualSeries = response.getBodyObject(SerialPublicationDto.class);
+        assertThat(actualSeries, is(equalTo(expectedSeries)));
+        var statusCode = response.getStatusCode();
+        assertThat(statusCode, is(equalTo(HTTP_OK)));
+        var contentType = response.getHeaders().get(CONTENT_TYPE);
+        assertThat(contentType, is(equalTo(expectedMediaType)));
+    }
+
+    @Test
+    void shouldReturnChannelIdWithRequestedYearIfThirdPartyDoesNotProvideYear() throws IOException {
```
Comment (truhacevkir) [inline src/test/java/no/sikt/nva/pubchannels/handler/fetch/BaseFetchSerialPublicationByIdentifierAndYearHandlerTest.java:112]: image

Can this test be also a part of `FetchByIdentifierAndYearHandlerTest`?

File: src/test/java/no/sikt/nva/pubchannels/handler/fetch/BaseFetchSerialPublicationByIdentifierAndYearHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+        var statusCode = response.getStatusCode();
+        assertThat(statusCode, is(equalTo(HTTP_OK)));
+
+        var actualSeries = response.getBodyObject(SerialPublicationDto.class);
+        assertThat(actualSeries, is(equalTo(expectedSeries)));
+    }
+
+    @Test
+    void shouldReturnRedirectWhenChannelRegistryReturnsRedirect() throws IOException {
```
Comment (truhacevkir) [inline src/test/java/no/sikt/nva/pubchannels/handler/fetch/BaseFetchSerialPublicationByIdentifierAndYearHandlerTest.java:132]: Same with this method, it looks like all three classes are using the same business method from `FetchByIdentifierAndYearHandler`

File: src/test/java/no/sikt/nva/pubchannels/handler/fetch/FetchByIdentifierAndYearHandlerTest.java [hunk 5]
```diff
@@ -1,25 +1,50 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
 import org.mockito.Mockito;
+import org.zalando.problem.Problem;
 
+// Common behaviour for FetchSerialPublicationByIdentifierAndYearHandler,
+// FetchPublisherByIdentifierAndYearHandler, FetchSeriesByIdentifierAndYearHandler and
+// FetchJournalByIdentifierAndYearHandler are tested here
```
Comment (truhacevkir) [inline src/test/java/no/sikt/nva/pubchannels/handler/fetch/FetchByIdentifierAndYearHandlerTest.java:47]: Maybe as javadoc

**Review by truhacevkir — APPROVED** [review]
Excellent, some comments regarding test methods that may be moved to common test class :)

File: src/test/java/no/sikt/nva/pubchannels/handler/fetch/BaseFetchSerialPublicationByIdentifierAndYearHandlerTest.java (hunk 3 above)
Comment (monaullah) [reply src/test/java/no/sikt/nva/pubchannels/handler/fetch/BaseFetchSerialPublicationByIdentifierAndYearHandlerTest.java:124]: In future

**Pull request merged**